### ISSUE TITLE
automatically install dump library

### ISSUE THREAD

**Issue by Almenon** [Almenon/AREPL-vscode#74]
Now that I have a setting for default imports, I just need to put in "from arepl import dump" and the arepl session will have a dump function automatically availible.  This could have been done earlier but when testing I ran into a problem - altough AREPL could evaluate the dump just fine, because it could import dump from the relative folder, normal python debugging would break.

To fix this I need to make sure the dump library is installed globally.  There are several options here:

1. Add file to PYTONPATH so it is automatically picked up.  Need to look more into this.
2. Copy files over to site packages directory (thought about doing this but kinda hacky, better to just do pip package)
3. pip package

**Comment by Almenon** [comment]
No matter what I will want the dump library hardcoded into either arepl-vscode or arepl-backend for offline installation.

https://stackoverflow.com/questions/11091623/python-packages-offline-installation

Online installation could potentially be slow - I would worry that installation would not finish by the time user opens arepl

**Comment by Almenon** [comment]
Also whatever way I go I'm gonna want to rename arepl.py to arepldump to avoid potential collision with a future pip library (AREPL electron or we service through pip)

**Comment by Almenon** [comment]
technically I could do the following:

```python
import sys;sys.path.append("C:\\Users\\anon\\.vscode\\extensions\\almenon.arepl-0.1.1\\node_modules\\arepl-backend\\python");from arepldump import dump
```

But it's way too long to properly go on one line :(

```python
import sys;sys.path.append("C:\\Users\\anon\\.vscode\\extensions\\almenon.arepl-0.1.1\\node_modules\\arepl-backend\\python");
from arepldump import dump
```

Broken up it's still a bit too long, but I don't want to insert any more than 1 line of code into the users file.

**Comment by Almenon** [comment]
I think I'll just write a .pth file with the path to the users site-packages directory.  Seems the simplest option.

**Comment by Almenon** [comment]
so somehow arepldump retains state across arepl's exec() calls..... not sure why *or how* the heck that is happening.

Looks like more debugging fun for me :(  

If this is the easy path of adding local scope functionality, i shudder to think how long the hard path (using bdb to basically make my own debugger) will take

**Comment by Almenon** [comment]
so I've got it working now but I'm not 100% happy with the result....

![image](https://user-images.githubusercontent.com/13080965/42136472-e3ec34ec-7d10-11e8-8f1a-645f323ff032.png)

There's no clear visual connection between the line being dumped and the output.  Something like wolf would be nice where I could show the values inline.  

The ordering is bad too - either it should be alphabeticall or chronological, but it is neither.

And although having dump output in JSON makes it very easy to display, the nesting levels become a problem.  AREPL only shows up to level 2 by default, so variables inside a dump are already at level 1 and can only show another level of nesting.  Hence why first-level lists in a dump are not expanded. To fix this I could either implement #75 or have the dumps be seperate groups of JSON, whether it be in the display or inline.
